code/modules/mob/living/simple_animal/abnormality/aleph/army_in_black.dm

```diff
@@ -0,0 +1,498 @@
+		H.adjustBruteLoss(-20)
+		H.adjustSanityLoss(20)
+	new /obj/effect/temp_visual/pink_explosion(get_turf(src))
+	qdel(src)
+
+//hostile breach mob
+/mob/living/simple_animal/hostile/army_enemy
+	name = "Army In Black"
+	desc = "Yes.. we, the Army in Black.. blend into the human heart.. and drive away good thoughts.."
+	icon = 'ModularTegustation/Teguicons/64x64.dmi'
+	icon_state = "armyinblack"
+	icon_living = "armyinblack"
+	icon_dead = "armyinblack_heart"
+	pixel_x = -16
+	base_pixel_x = -16
+	faction = list("hostile")
+	attack_verb_continuous = "attacks"
+	attack_verb_simple = "attack"
+	attack_sound = 'sound/abnormalities/armyinblack/pink_heal.ogg'
+	/*Core stats*/
+	health = 900
+	maxHealth = 900
+	obj_damage = 50
+	damage_coeff = list(BRUTE = 1, RED_DAMAGE = 1.2, WHITE_DAMAGE = 0.6, BLACK_DAMAGE = 1.0, PALE_DAMAGE = 0.8)
+	ranged = TRUE
+	minimum_distance = 2
+	speed = 2
+	move_to_delay = 20//TODO: make this determined by how far away it spawned from the beacon
+	stat_attack = HARD_CRIT
+	del_on_death = FALSE
+	density = FALSE
+	is_flying_animal = TRUE
+	var/list/fear_affected = list()
+	var/can_act = TRUE
+	var/shot_cooldown
+	var/shot_cooldown_time = 5 SECONDS
+	var/boom_damage = 70
+	var/boom_radius = 20
+	var/targetted_beacon
+	var/list/moving_path
+
+//movement and AI
+/mob/living/simple_animal/hostile/army_enemy/AttackingTarget()
+	return
+
+/mob/living/simple_animal/hostile/army_enemy/proc/GoToBeacon()
+	if(QDELETED(src))
+		return
+	patrol_to(get_turf(targetted_beacon))//this causes a runtime when the unit is deleted due to invoking asynchronous move_to! Please help!
+	return
+
+/mob/living/simple_animal/hostile/army_enemy/Life()
+	. = ..()
+	if(!.)
+		return
+	if(targetted_beacon)
+		GoToBeacon()
+	return
+
+/mob/living/simple_animal/hostile/army_enemy/Initialize()
+	..()
+	addtimer(CALLBACK(src, .proc/Explode), 120 SECONDS)
+	var/list/depts = shuffle(GLOB.department_centers)
+	if(!LAZYLEN(depts))
+		return
+	for (var/turf/T in depts)
+		if(!(get_area(T) in (get_areas_in_range(20, get_area(src)))))//if they're not within 20 tiles of T, pick the next one.
+			continue//FIXME: the final element should forcibly attempt to spawn these
+		if(/obj/effect/pink_beacon in T.contents)//don't spawn 2 in the same dept(FIXME: doesn't work, even with sleep() called on SpawnAdds!)
+			continue
+		var/obj/effect/pink_beacon/P = new(get_turf(T))//beacon
+		targetted_beacon = P
+		P.targetted_army = src
+		INVOKE_ASYNC(src, .proc/SetSpeed)
+		break
+
+/mob/living/simple_animal/hostile/army_enemy/death()
+	animate(src, alpha = 0, time = 10 SECONDS)
+	QDEL_IN(src, 10 SECONDS)
+	..()
+
+/mob/living/simple_animal/hostile/army_enemy/proc/SetSpeed()
+	var/dist_travelled = get_path_to(src, get_turf(targetted_beacon), /turf/proc/Distance_cardinal, 0, 200)
+	move_to_delay -= clamp((LAZYLEN(dist_travelled) / 4), 0, 15)//armies that spawn closest to dept centers can actually be suppressed this way, while further ones remain a threat. Math needs tweaking
```

Comment:
Consider, if you're gonna do this, why not use the existing patrol_path?

Reply:
ditto, I don't know how patrol_path works and merely looked at hostile.dm in attempts to find a relevant proc